app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -1,76 +1,363 @@
+            binding.userProfileName.text = user.displayName
+            binding.userProfileEmail.text = "Email: " + user.email
+
+            userProfileViewModel.userObject.observe(this, Observer {
+                binding.userProfileBio.text = userProfileViewModel.userObject.value?.userObject?.bio
+                binding.userProfileBioEdit.setText(userProfileViewModel.userObject.value?.userObject?.bio)
+                val countryCode = userProfileViewModel.userObject.value?.userObject?.countryCode
+                val phone = userProfileViewModel.userObject.value?.userObject?.phone
+
+                if (phone != null) {
+                    binding.userProfileAddPhoneButton.visibility = View.GONE
+                    binding.userProfilePhoneLayout.visibility = View.VISIBLE
+                    binding.userProfilePhone.text = ("Phone: +"
+                            + countryCode
+                            + " "
+                            + phone)
+                }
+            })
+
+        } else // If the google account does not exist, then it means the user logged in with their firebase account
         {
             // Get the firebase user from FireBaseAuth
             val user = FirebaseAuth.getInstance().currentUser
-            user?.let{
+            user?.let {
 
                 // Set the full name, email on the profile from the firebase account, and show the placeholder pfp as the pfp
-                binding.userProfileName.setText(user.displayName)
-                binding.userProfileEmail.setText("Email: " + user.email)
+
+                binding.userProfileName.text = user.displayName
+                binding.userProfileEmail.text = "Email: " + user.email
                 binding.userProfilePicture.setImageResource(R.drawable.user_pfp_example)
-            }
 
+                userProfileViewModel.userObject.observe(this, Observer {
+                    binding.userProfileBio.text =
+                        userProfileViewModel.userObject.value?.userObject?.bio
+                    binding.userProfileBioEdit.setText(userProfileViewModel.userObject.value?.userObject?.bio)
+
+                    val countryCode = userProfileViewModel.userObject.value?.userObject?.countryCode
+                    val phone = userProfileViewModel.userObject.value?.userObject?.phone
+
+                    if (phone != null) {
+                        binding.userProfileAddPhoneButton.visibility = View.GONE
+                        binding.userProfilePhoneLayout.visibility = View.VISIBLE
+                        binding.userProfilePhone.text = ("Phone: +"
+                                + countryCode
+                                + " "
+                                + phone)
+                    }
+                })
+            }
         }
 
         // Create Action Bar var so we can 1) display it with a proper title and 2) put a working back button on it
         var actionBar = supportActionBar
```

Comment:
Put val instead of var, I keep repeating the same comments which makes me think you are not addressing them properly and thoroughly enough